### ISSUE TITLE
Sort_by function not work very well with hungarian letters

### ISSUE THREAD

**Issue by og900aero** [sxyazi/yazi#1092]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

st 0.9.1

### Did you try the latest code to see if this problem got fixed?

Tried, but the problem still

### `yazi --debug` output

<details>

```Yazi
    Version: 0.2.5 (53d658d 2024-05-26)
    OS: linux-x86_64 (unix)
    Debug: false

Ya
    Version: Ok(Output { status: ExitStatus(unix_wait_status(0)), stdout: "Ya 0.2.5 (53d658d 2024-05-26)\n", stderr: "" })

Emulator
    Emulator.via_env: ("st-256color", "")
    Emulator.via_csi: Ok(Unknown([]))
    Emulator.detect: Unknown([])

Adaptor
    Adaptor.matches: X11

Desktop
    XDG_SESSION_TYPE: Some("tty")
    WAYLAND_DISPLAY: None
    DISPLAY: Some(":0")

SSH
    shared.in_ssh_connection: false

WSL
    /proc/sys/fs/binfmt_misc/WSLInterop: false

Variables
    SHELL: Some("/bin/bash")
    EDITOR: Some("micro")
    ZELLIJ_SESSION_NAME: None
    YAZI_FILE_ONE: None
    YAZI_CONFIG_HOME: None

file(1)
    Version: Ok(Output { status: ExitStatus(unix_wait_status(0)), stdout: "file-5.44\nmagic file from /etc/magic:/usr/share/misc/magic\n", stderr: "" })

Text Opener
    default: Some(Opener { run: "$EDITOR \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block: Some(Opener { run: "$EDITOR \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

tmux
    TMUX: false

Ueberzug++
    Version: Ok(Output { status: ExitStatus(unix_wait_status(0)), stdout: "ueberzugpp 2.9.5\n", stderr: "" })

```

</details>

### Describe the bug

Alignment of accented characters does not yet work properly.

![screenshot_1716747075](https://github.com/sxyazi/yazi/assets/49487376/e52a19ae-59dc-477a-b320-8787c9efc605)

Here you can see that it doesn't handle accented characters properly because it actually sorts by the second letter.

### Expected Behavior

The correct sequence is always the following regardless of what the second character contains

A....
Á....
...
E
É
...
I
Í
...
O
Ó
Ö
Ő
...
U
Ú
Ü
Ű
...

### To Reproduce

![screenshot_1716747075](https://github.com/sxyazi/yazi/assets/49487376/934d3ed7-8a79-438a-82d4-960f4c27d99a)

**Comment by sxyazi** [comment]
You need to enable it in your `yazi.toml` like this:

```toml
[manager]
sort_by              = "natural"
sort_transliteration = true
```

Note: I just resubmitted the commit https://github.com/sxyazi/yazi/commit/1ab3df6850640c034a624566abcc8ef968c77a89 and renamed `sort_transliteration` to `sort_translit` because it was too long lol.

See https://github.com/sxyazi/yazi/pull/1053 for more details.

**Comment by og900aero** [comment]
> ```toml
> sort_by              = "natural"
> sort_transliteration = true
> ```

I did it before make this issue, and does not yet work properly.

**Comment by sxyazi** [comment]
Weird, it works for me.

![screenshot-000062](https://github.com/sxyazi/yazi/assets/17523360/eb3f5634-8dc1-412d-a226-401529a5d0b5)


Could you record a gif to show me your steps? Include the Yazi version, configuration file, and sorting operation.

**Comment by og900aero** [comment]
The example you show only works because the second letter of words that start with accented letters is further away.
Create the files with the following names, and then it won't be good for you too:

ábrahám
aladár
balassagyarmat
édrián
elemér
ubul
údrira

**Comment by sxyazi** [comment]
Is this the correct order?

![screenshot-000063](https://github.com/sxyazi/yazi/assets/17523360/e8607efe-386c-44f7-af48-7e9c686725f2)

**Comment by og900aero** [comment]
> Is this the correct order?
> 
> ![screenshot-000063](https://private-user-images.githubusercontent.com/17523360/334089799-e8607efe-386c-44f7-af48-7e9c686725f2.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4MTEwNzcsIm5iZiI6MTcxNjgxMDc3NywicGF0aCI6Ii8xNzUyMzM2MC8zMzQwODk3OTktZTg2MDdlZmUtMzg2Yy00NGY3LWFmNDgtN2U5YzY4NjcyNWYyLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI3VDExNTI1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllNGY4NTk1NWQ0Y2JjNTBmNzFiZmMxZjY5YWNiNmEzNzBiNjRmNTVmNGI3ZGZmZWQ5Nzk1ZDI1OTk2ZTZmOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KDqGn2jciasmVf7cHcl2B_bBFeCd6vwADqSnIn6bLO8)

No. The correct sequence is always the following regardless of what the second character contains:
A....
Á....
...
E
É
...
I
Í
...
O
Ó
Ö
Ő
...
U
Ú
Ü
Ű
...

So:
aladár
ábrahám
balassagyarmat
elemér
édrián
ubul
údrira

**Comment by sxyazi** [comment]
This is how `sort_translit` works: when you enable it, it removes diacritical marks from characters in a string, like converting `Á` to `A` for sorting purposes. This means `Á` and `A` are treated as equivalent and sorted according to the general character order, grouping similar characters together. 

I'm not sure if there's a better way to achieve this.

**Comment by og900aero** [comment]
Accented characters (such as á, é, í, ó, ö, ő, ú, ű) are not equivalent to the unaccented versions. Each language has its own accented characters, and they follow a given rule.

**Comment by sxyazi** [comment]
If different languages have different processing rules, it would be more complex. 

Currently, Yazi has a built-in [character mapping table](https://github.com/sxyazi/yazi/blob/main/yazi-shared/src/translit/table.rs) that simply replaces characters without considering the differences between languages.
